app/models/concerns/earliest_date_validator.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Validates that a given date (attribute) does not occur before the date provided in options
+class EarliestDateValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return unless options.key?(:date)
```

Comment:
This is likely just me not being as familiar with concerns in Rails yet, but where is `options` defined or passed?

Reply:
Options is a hash that you pass to an `EachValidator` which you can see happens in the model... `earliest_date: { options }` by `return unless` the options that must be present are established.